modules/core/src/main/java/org/apache/ignite/internal/processors/task/TaskExecutionOptions.java

```diff
@@ -176,13 +176,13 @@ public TaskExecutionOptions asSystemTask() {
     }
 
     /** */
-    public boolean isAuthenticationDisabled() {
-        return isAuthDisabled;
+    public boolean isSystemTaskGuard() {
+        return isSystemTaskGuard;
     }
 
     /** */
-    public TaskExecutionOptions withAuthenticationDisabled() {
-        isAuthDisabled = true;
+    public TaskExecutionOptions withSystemTaskGuard() {
+        isSystemTaskGuard = true;
 
         return this;
     }
```

Comment:
Looks like, this spetifies option for current thread only, is it safe?

Reply:
The option itself is not thread local. 
Take a look at IgniteComputeImpl, please. We do use TaskExecutionOptions as thread local variable here, but isPublicFacade flag is stored separately and applied to each compute execution method. So it is safe.